homework 1/ReaderWriterLock/SharedResourceRwLock.cs

```diff
@@ -1,19 +1,38 @@
+using System.Threading;
+
 namespace ReaderWriterLock;
 
 public class SharedResourceRwLock : SharedResourceBase
 {
+    private readonly ReaderWriterLockSlim _lockSlim = new();
+    private string _content;
+    
     public override void Write(string data)
     {
-        throw new System.NotImplementedException();
+        _lockSlim.EnterWriteLock();
+        _content = data;
+        _lockSlim.ExitWriteLock();
```

Comment:
лучше делать
```suggestion
        _lockSlim.EnterWriteLock();
        try
        {
            _content = data;
        }
        finally
        {
            _lockSlim.ExitWriteLock();
        }
```
чтобы даже в случае ошибки гарантированно освободить лок